lib/imap.c

```diff
@@ -663,8 +663,54 @@ static CURLcode imap_perform_list(struct Curl_easy *data)
     free(mailbox);
   }
 
-  if(!result)
-    imap_state(data, IMAP_LIST);
+  if(!result) {
+    bool isFetch = FALSE;
+
+    /* Switch state to be FETCH if custom command is a fetch command */
+    /* so output is handled correctly. */
+
+    if(imap->custom) {
+      if(strncasecompare(imap->custom, "FETCH", 5)) {
+        isFetch = true;
+      }
+
+      /* e.g. "UID FETCH 107 BODY.PEEK[]" */
+
+      else if(strncasecompare(imap->custom, "UID", 3) &&
+              strncasecompare(imap->custom_params, " FETCH", 6)) {
+        isFetch = true;
+      }
+
+      if(isFetch) {
+        /*
+          if we have FETCH with range, we need a LIST state,
+          otherwise FETCH state, e.g.
+          FETCH 175 BODY.PEEK[]
+          to get one email
+          vs.
+          UID FETCH 1:* (FLAGS INTERNALDATE RFC822.SIZE
+          BODY.PEEK[HEADER.FIELDS (Message-Id DATE FROM
+          SUBJECT TO SENDER REPLY-TO CC BCC)])
+          to get the list of emails
+        */
+        const char *custom_params =
+          imap->custom_params ? imap->custom_params : "";
+        const char *positionColon = strchr(custom_params, ':');
```

Comment:
```suggestion
        const char *positionColon = strchr(imap->custom_params, ':');
```

... when combined with the change above.